astrix-fault-tolerance/src/main/java/com/avanza/astrix/ft/hystrix/HystrixFaultToleranceFactory.java

```diff
@@ -17,83 +17,111 @@
-		this.hystrixCommandKeyFactory = new HystrixCommandKeyFactory(id, hystrixCommandNamingStrategy);
-	}
-	
-	@Override
-	public BeanFaultToleranceMetricsMBean createBeanFaultToleranceMetricsMBean(AstrixBeanKey<?> beanKey) {
-		return new BeanFaultToleranceMetrics(getCommandKey(beanKey), getThreadPoolKey(beanKey));
-	}
-	
-	HystrixCommandGroupKey getGroupKey(AstrixBeanKey<?> beanKey) {
-		HystrixCommandGroupKey result = hystrixCommandKeyFactory.createGroupKey(beanKey);
-		this.beanMapping.registerBeanKey(result.name(), beanKey);
-		return multiConfigId.encode(result);
-	}
+    /*
+     * Astrix allows multiple AstrixContext within the same JVM, although
+     * more than one AstrixContext within a single JVM is most often only used during unit testing. However,
+     * since Astrix allows multiple AstrixContext's we want those to be isolated from each other. Hystrix only
+     * allows registering a global strategy for each exposed SPI. Therefore Astrix registers a global "dispatcher"
+     * strategy (see HystrixPluginDispatcher) that dispatches each invocation to a HystrixPlugin to
+     * the HystrixPlugin instance associated with a given AstrixContext.
+     *
+     * The "id" property is used by the dispatcher to identify the HystrixPlugin instances associated with a given
+     * AstrixContext.
+     */
+
+    private static final AtomicInteger idGenerator = new AtomicInteger(0);
+
+    private final BeanMapping beanMapping;
+    private final HystrixCommandKeyFactory hystrixCommandKeyFactory;
+    private final String id;
+    private final ContextPropagation contextPropagation;
+    private MultiConfigId multiConfigId = MultiConfigId.create("astrix");
+
+    /**
+     * @deprecated please use {@link #HystrixFaultToleranceFactory(HystrixCommandNamingStrategy, AstrixConcurrencyStrategy, BeanConfigurationPropertiesStrategy, BeanMapping, AstrixTraceProvider, AstrixConfig)}
+     */
+    @Deprecated
+    public HystrixFaultToleranceFactory(HystrixCommandNamingStrategy hystrixCommandNamingStrategy,
+                                        AstrixConcurrencyStrategy concurrencyStrategy,
+                                        BeanConfigurationPropertiesStrategy propertiesStrategy,
+                                        BeanMapping beanMapping,
+                                        AstrixConfig config) {
+        this(
+                hystrixCommandNamingStrategy,
+                concurrencyStrategy,
+                propertiesStrategy,
+                beanMapping,
+                new DefaultTraceProvider(),
+                config
+        );
+    }
+
+    @AstrixInject
+    public HystrixFaultToleranceFactory(
+            HystrixCommandNamingStrategy hystrixCommandNamingStrategy,
+            AstrixConcurrencyStrategy concurrencyStrategy,
+            BeanConfigurationPropertiesStrategy propertiesStrategy,
+            BeanMapping beanMapping,
+            AstrixTraceProvider astrixTraceProvider,
```

Comment:
AstrixTraceProvider should be added as the last argument to HystrixFaultToleranceFactory constructor to match other constructor

Reply:
👍  done